### PR TITLE
Test cleanup based on team code styles

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/AmqpConnectionPoolSettings.cs [hunk 1]
```diff
@@ -1,81 +1,96 @@
...
             }
         }
 
+        /// <summary>
+        /// Whether or not to use connection pooling
+        /// </summary>
         public bool Pooling { get; set; }
 
+        /// <summary>
+        ///
```
Comment (abhipsaMisra) [inline iothub/device/src/AmqpConnectionPoolSettings.cs]: empty summary?

Comment (drwill-ms) [reply iothub/device/src/AmqpConnectionPoolSettings.cs]: I probably paused to inspect what it really was, and didn't get back to it. What is it, exactly?

Comment (abhipsaMisra) [reply iothub/device/src/AmqpConnectionPoolSettings.cs]: The name would suggest that it is being used to validate client side connection heartbeat interval, and gracefully close the connection if no frames are sent for the threshold specified. However, @davilu recently added another property called 'IdleTimeout`, which seems to be targeting similar behavior.
@davilu - is there a different use case for `ConnectionIdleTimeout`? I cannot find it referenced anywhere explicitly, other than in the `Equals` method below.

Comment (drwill-ms) [reply iothub/device/src/AmqpConnectionPoolSettings.cs]: I don't see the property being referenced outside the class except by a test method. Am I missing something?

File: iothub/device/src/Http1TransportSettings.cs [hunk 2]
```diff
@@ -26,7 +26,12 @@ public TransportType GetTransportType()
         }
 
         public X509Certificate2 ClientCertificate { get; set; }
-        public TimeSpan DefaultReceiveTimeout => DefaultOperationTimeout;
+
+        public TimeSpan GetDefaultReceiveTimeout()
+        {
+            return DefaultOperationTimeout;
```
Comment (bikamani) [inline iothub/device/src/Http1TransportSettings.cs]: so why not use a get/set here?

File: iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs [hunk 3]
```diff
@@ -61,7 +61,7 @@ public MqttTransportSettings(TransportType transportType)
             this.QoSPropertyName = "mqtt-qos";
             this.RetainPropertyName = "mqtt-retain";
             this.WillMessage = null;
-            this.DefaultReceiveTimeout = TimeSpan.FromSeconds(DefaultReceiveTimeoutInSeconds);
+            this.SetDefaultReceiveTimeout(TimeSpan.FromSeconds(DefaultReceiveTimeoutInSeconds));
```
Comment (bikamani) [inline iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs]: remove all "this."?

File: iothub/device/src/Transport/RetryDelegatingHandler.cs [hunk 4]
```diff
@@ -11,51 +11,47 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport
 {
-    using ExponentialBackoff = Microsoft.Azure.Devices.Client.ExponentialBackoff;
-
     internal class RetryDelegatingHandler : DefaultDelegatingHandler
     {
         // RetryCount is used for testing purpose and is equal to MaxValue in prod.
-        private static int s_retryMaxCount = int.MaxValue;
+        private const int RetryMaxCount = int.MaxValue;
```
Comment (bikamani) [inline iothub/device/src/Transport/RetryDelegatingHandler.cs]: shouldn't this be _camelCase?

File: iothub/device/src/Transport/RetryDelegatingHandler.cs [hunk 5]
```diff
@@ -74,18 +70,21 @@ public override async Task SendEventAsync(Message message, CancellationToken can
...
+                    .ExecuteAsync(
+                        async () =>
+                        {
+                            await EnsureOpenedAsync(cancellationToken).ConfigureAwait(false);
 
-                    if (message.IsBodyCalled)
-                    {
-                        message.ResetBody();
-                    }
+                            if (message.IsBodyCalled)
```
Comment (bikamani) [inline iothub/device/src/Transport/RetryDelegatingHandler.cs]: is it spaces instead of tab here?

File: iothub/device/src/Transport/RetryDelegatingHandler.cs [hunk 6]
```diff
@@ -513,52 +557,56 @@ private async Task EnsureOpenedAsync(TimeoutHelper timeoutHelper)
...
-                    HandleConnectionStatusExceptions(ex);
-                    throw;
-                }
-                finally
-                {
-                    if (Logging.IsEnabled) Logging.Exit(this, cancellationToken, nameof(OpenAsync));
-                }
-            },
-            cancellationToken);
+                            // Will throw on error.
```
Comment (bikamani) [inline iothub/device/src/Transport/RetryDelegatingHandler.cs]: again this shows up as spaces?

File: iothub/device/tests/TransportSettingsTests.cs [hunk 7]
```diff
@@ -1,271 +1,333 @@
...
-    using Microsoft.Azure.Devices.Client.Transport.Mqtt;
-    using Microsoft.VisualStudio.TestTools.UnitTesting;
-
     [TestClass]
     [TestCategory("Unit")]
     public class TransportSettingsTests
     {
-        const string LocalCertFilename = "..\\..\\Microsoft.Azure.Devices.Client.Test\\LocalNoChain.pfx";
-        const string LocalCertPasswordFile = "..\\..\\Microsoft.Azure.Devices.Client.Test\\TestCertsPassword.txt";
+        private const string LocalCertFilename = "..\\..\\Microsoft.Azure.Devices.Client.Test\\LocalNoChain.pfx";
```
Comment (bikamani) [inline iothub/device/tests/TransportSettingsTests.cs]: _camelCase

File: iothub/device/tests/TransportSettingsTests.cs [hunk 8]
```diff
@@ -1,271 +1,333 @@
...
+        private const string LocalCertPasswordFile = "..\\..\\Microsoft.Azure.Devices.Client.Test\\TestCertsPassword.txt";
 
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         [TestMethod]
-        public void TransportSettingsTestTransportTypeAmqp()
+        [ExpectedException(typeof(ArgumentOutOfRangeException))]
+        public void AmqpTransportSettings_InvalidTransportTypeAmqp()
```
Comment (bikamani) [inline iothub/device/tests/TransportSettingsTests.cs]: For the naming here, did you intend to mention Amqp twice?

Same applies for tests below

File: iothub/device/tests/TransportSettingsTests.cs [hunk 9]
```diff
@@ -1,271 +1,333 @@
...
+        public void AmqpTransportSettings_UnderPrefetchCountMin()
         {
-            var amqpConnectionPoolSettings = new AmqpConnectionPoolSettings();
-            var transportSetting = new AmqpTransportSettings(TransportType.Amqp_Tcp_Only, 0, amqpConnectionPoolSettings);
+            _ = new AmqpTransportSettings(TransportType.Amqp_Tcp_Only, 0, new AmqpConnectionPoolSettings());
         }
 
         [TestMethod]
-        public void TransportSettingsTestTransportTypeAmqpWebSocket()
+        public void AmqpTransportSettings_Defaults()
```
Comment (bikamani) [inline iothub/device/tests/TransportSettingsTests.cs]: Can there be a better naming for this test?

File: iothub/device/src/ITransportSettings.cs [hunk 10]
```diff
@@ -19,6 +19,6 @@ public interface ITransportSettings
         /// <summary>
         /// The default receive timeout.
         /// </summary>
-        TimeSpan DefaultReceiveTimeout { get; }
```
Comment (abhipsaMisra) [inline iothub/device/src/ITransportSettings.cs]: Wouldn't this qualify as a breaking API change?

File: iothub/device/src/TelemetryMethods.cs [hunk 11]
```diff
@@ -26,7 +26,10 @@ public static string GetSqmMachineId()
                 catch (Exception ex)
                 {
                     Debug.Assert(false, ex.Message);
-                    if (Logging.IsEnabled) Logging.Error(null, ex, nameof(TelemetryMethods));
```
Comment (abhipsaMisra) [inline iothub/device/src/TelemetryMethods.cs]: Isn't a single statement allowed to be without enclosing blocks?

File: iothub/device/tests/Microsoft.Azure.Devices.Client.Tests.csproj [hunk 12]
```diff
@@ -12,6 +14,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\..\..\provisioning\service\src\Microsoft.Azure.Devices.Provisioning.Service.csproj" />
```
Comment (abhipsaMisra) [inline iothub/device/tests/Microsoft.Azure.Devices.Client.Tests.csproj]: Why is this required?

File: iothub/device/src/Transport/RetryDelegatingHandler.cs (hunk 4 above)
Comment (drwill) [reply iothub/device/src/Transport/RetryDelegatingHandler.cs]: Const are TitleCase. I never liked that standard, but that is what most teams seem to go with.

File: iothub/device/src/Transport/RetryDelegatingHandler.cs (hunk 5 above)
Comment (drwill) [reply iothub/device/src/Transport/RetryDelegatingHandler.cs]: yes

File: iothub/device/src/TelemetryMethods.cs (hunk 11 above)
Comment (drwill) [reply iothub/device/src/TelemetryMethods.cs]: Allowed by the language, but not preferred by the style. I'm going to revert this change for now, because I plan to put the condition in the method instead.

File: iothub/device/src/Transport/RetryDelegatingHandler.cs (hunk 6 above)
Comment (drwill) [reply iothub/device/src/Transport/RetryDelegatingHandler.cs]: Tabbing in based on newlines in 560-562

File: iothub/device/tests/Microsoft.Azure.Devices.Client.Tests.csproj (hunk 12 above)
Comment (drwill) [reply iothub/device/tests/Microsoft.Azure.Devices.Client.Tests.csproj]: Not sure how it got in there. I'm removing it.

File: iothub/device/tests/TransportSettingsTests.cs (hunk 7 above)
Comment (drwill) [reply iothub/device/tests/TransportSettingsTests.cs]: not for const. The editorconfig file has the rules in it, and the IDE will suggest when the name needs to change.

File: iothub/device/tests/TransportSettingsTests.cs (hunk 8 above)
Comment (drwill) [reply iothub/device/tests/TransportSettingsTests.cs]: To disambiguate between Amqp and AmpqHttp

File: iothub/device/tests/TransportSettingsTests.cs (hunk 9 above)
Comment (drwill) [reply iothub/device/tests/TransportSettingsTests.cs]: How's AmqpTransportSettings_DefaultPropertyValues?

File: iothub/device/src/Http1TransportSettings.cs (hunk 2 above)
Comment (drwill) [reply iothub/device/src/Http1TransportSettings.cs]: Reverted

File: iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs (hunk 3 above)
Comment (drwill) [reply iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs]: I haven't made a full pass on this file, but this one was easy to do, so I did it.

File: iothub/device/src/ITransportSettings.cs (hunk 10 above)
Comment (drwill) [reply iothub/device/src/ITransportSettings.cs]: Yes, I didn't mean for that change to happen. Reverted.

File: iothub/device/tests/RetryPolicyTests.cs [hunk 13]
```diff
@@ -1,26 +1,24 @@
...
-        public void NoRetryPolicy_VerifyBehavior_Success()
+        public void NoRetryPolicy_RecommendsNo()
         {
+            // arrange
             var noRetryPolicy = new NoRetry();
 
-            TimeSpan retryInterval;
-            Assert.IsFalse(noRetryPolicy.ShouldRetry(Arg.Any<int>(), Arg.Any<Exception>(), out retryInterval));
+            // act and assert
+            Assert.IsFalse(noRetryPolicy.ShouldRetry(0, null, out TimeSpan retryInterval));
```
Comment (abhipsaMisra) [inline iothub/device/tests/RetryPolicyTests.cs]: Isn't `Any<Exception>` different from null?

File: iothub/device/tests/TransportSettingsTests.cs [hunk 14]
```diff
@@ -1,271 +1,333 @@
...
         }
 
-        [ExpectedException(typeof(ArgumentOutOfRangeException))]
         [TestMethod]
-        public void TransportSettingsTestTransportTypeAmqpHttp()
+        [ExpectedException(typeof(ArgumentOutOfRangeException))]
+        public void AmqpTransportSettings_InvalidTransportTypeAmqpHttp()
         {
-            var transportSetting = new AmqpTransportSettings(TransportType.Http1);
+            _ = new AmqpTransportSettings(TransportType.Http1);
```
Comment (abhipsaMisra) [inline iothub/device/tests/TransportSettingsTests.cs]: cool!

File: iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs [hunk 15]
```diff
@@ -94,10 +96,15 @@ public MqttTransportSettings(TransportType transportType)
 
         public TransportType GetTransportType()
         {
-            return this.transportType;
+            return _transportType;
         }
 
-        public TimeSpan DefaultReceiveTimeout { get; set; }
```
Comment (abhipsaMisra) [inline iothub/device/src/Transport/Mqtt/MqttTransportSettings.cs]: We're still changing the public API; we cannot set DefaultReceiveTimeout  anymore

File: iothub/device/tests/RetryPolicyTests.cs (hunk 13 above)
Comment (drwill-ms) [reply iothub/device/tests/RetryPolicyTests.cs]: turns out, no
